src/Tribe/Attendee_Registration/Rewrite.php

```diff
@@ -0,0 +1,158 @@
+
+	/**
+	 * Tribe__Tickets__Attendee_Registration__Rewrite constructor.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Rewrite|null $wp_rewrite
+	 */
+	public function __construct( WP_Rewrite $wp_rewrite = null ) {
+		$this->rewrite = $wp_rewrite;
+	}
+
+	/**
+	 * Generate the Rewrite Rules
+	 *
+	 * @since TBD
+	 *
+	 * @param  WP_Rewrite $wp_rewrite WordPress Rewrite that will be modified, pass it by reference (&$wp_rewrite)
+	 */
+	public function filter_generate( WP_Rewrite $wp_rewrite ) {
+		$this->setup( $wp_rewrite );
+
+		/**
+		 * Use this to change the Tribe__Tickets__Attendee_Registration__Rewrite instance before new rules
+		 * are committed.
+		 *
+		 * Should be used when you want to add more rewrite rules without having to
+		 * deal with the array merge, noting that rules for Event Tickets are
+		 * themselves added via this hook (default priority).
+		 *
+		 * @var Tribe__Tickets__Attendee_Registration__Rewrite $rewrite
+		 */
+		do_action( 'tribe_tickets_pre_rewrite', $this );
+
+		/**
+		 * Provides an opportunity to modify Event Tickets' rewrite rules before they
+		 * are merged in to WP's own rewrite rules.
+		 *
+		 * @param array $events_rewrite_rules
+		 * @param Tribe__Tickets__Attendee_Registration__Rewrite $tribe_rewrite
+		 * @param WP_Rewrite $wp_rewrite WordPress Rewrite that will be modified.
+		 */
+		$this->rules = apply_filters( 'tribe_tickets_rewrite_rules_custom', $this->rules, $this, $wp_rewrite );
+
+		$wp_rewrite->rules = $this->rules + $wp_rewrite->rules;
+	}
+
+	/**
+	 * Sets up the rules required by Event Tickets.
+	 *
+	 * This should be called during tribe_tickets_pre_rewrite, which means other plugins needing to add rules
+	 * of their own can do so on the same hook at a lower or higher priority, according to how specific
+	 * those rules are.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Tickets__Attendee_Registration__Rewrite $rewrite
+	 */
+	public function generate_core_rules( Tribe__Tickets__Attendee_Registration__Rewrite $rewrite ) {
+		$rewrite->add( array( '{{ ' . Tribe__Tickets__Attendee_Registration__Main::QUERY_VAR . ' }}' ), array( Tribe__Tickets__Attendee_Registration__Main::QUERY_VAR => 1 ) );
+	}
+
+	/**
+	 * Add attendee-registration rewrite tag.
+	 *
+	 * @since TBD
+	 */
+	public function add_rewrite_tags() {
+		add_rewrite_tag( '%' . Tribe__Tickets__Attendee_Registration__Main::QUERY_VAR . '%', '([^&]+)' );
+	}
+
+	/**
+	 * Get the base slugs for the Plugin Rewrite rules
+	 *
+	 * WARNING: Don't mess with the filters below if you don't know what you are doing
+	 *
+	 * @since TBD
+	 *
+	 * @param  string $method Use "regex" to return a Regular Expression with the possible Base Slugs using l10n
+	 * @return object         Return Base Slugs with l10n variations
+	 */
+	public function get_bases( $method = 'regex' ) {
+		$tickets = tribe( 'tickets.main' );
+
+		/**
+		 * If you want to modify the base slugs before the i18n happens filter this use this filter
```

Comment:
"If you want to modify the base slugs before the i18n happens ~filter this~ use this filter"